spec/services/attribute_asserter_spec.rb

```diff
@@ -0,0 +1,28 @@
+require 'rails_helper'
+
+describe AttributeAsserter do
+  include SamlAuthHelper
+
+  let(:user) do
+    user = create(:user, :signed_up)
+    user.profiles.create(
+      active: true,
+      activated_at: Time.current,
+      verified_at: Time.current,
+      first_name: 'Jane'
+    )
+    user
+  end
+  let(:service_provider) { ServiceProvider.new(sp1_saml_settings.issuer) }
+  let(:subject) { described_class.new(user, service_provider, sp1_authnrequest) }
+
+  describe '#build' do
+    before do
+      subject.build
+    end
+
+    it 'sets user asserted_attributes' do
+      expect(user.asserted_attributes).to have_key :first_name
```

Comment:
Maybe this test would be better (clearer) if we mocked the ServiceProvider completely, with explicit attribute_bundle definition. I'll try that.